ci/scripts/deterministic-it-test.sh

```diff
@@ -16,6 +16,7 @@ mv target/ci-sim target/sim
 echo "--- Run integration tests in deterministic simulation mode"
 seq $TEST_NUM | parallel MADSIM_TEST_SEED={} NEXTEST_PROFILE=ci-sim \
  cargo nextest run \
+ --no-capture \
```

Comment:
Is there some reason for adding `no-capture`? IMO it should just be used locally, to observe realtime stream of events.

In CI, this should be disabled, so we only receive logs from failing tests.